app/src/main/java/com/example/parkingsystem/mvp/model/ParkingSpaceReservationModel.java

```diff
@@ -1,50 +1,71 @@
 package com.example.parkingsystem.mvp.model;
 
+import com.example.parkingsystem.database.ParkingSpaceReservationDB;
+import com.example.parkingsystem.entity.Reservation;
 import com.example.parkingsystem.mvp.contracts.ParkingSpaceReservationContract;
 
 public class ParkingSpaceReservationModel implements ParkingSpaceReservationContract.ParkingSpaceReservationModel {
-    private String dateStart;
-    private String timeStart;
-    private String dateEnd;
-    private String timeEnd;
+    private Reservation reservation = new Reservation();
+    private ParkingSpaceReservationDB database;
+
+    public ParkingSpaceReservationModel(ParkingSpaceReservationDB database) {
+        this.database = database;
+    }
+
+    @Override
+    public String getDateStart() {
+        return this.reservation.getDateStart();
+    }
 
     @Override
     public void setDateStart(String date) {
-        this.dateStart = date;
+        this.reservation.setDateStart(date);
     }
 
     @Override
-    public void setDateEnd(String date) {
-        this.dateEnd = date;
+    public String getTimeStart() {
+        return reservation.getTimeStart();
     }
 
     @Override
-    public String getDateStart() {
-        return dateStart;
+    public void setTimeStart(String time) {
+        this.reservation.setTimeStart(time);
     }
 
     @Override
     public String getDateEnd() {
-        return dateEnd;
+        return reservation.getDateEnd();
     }
 
     @Override
-    public void setTimeStart(String time) {
-        this.timeStart = time;
+    public void setDateEnd(String date) {
+        this.reservation.setDateEnd(date);
+    }
+
+    @Override
+    public String getTimeEnd() {
+        return reservation.getTimeEnd();
     }
 
     @Override
     public void setTimeEnd(String time) {
-        this.timeEnd = time;
+        this.reservation.setTimeEnd(time);
     }
 
     @Override
-    public String getTimeStart() {
-        return timeStart;
+    public void setParkingSpace(int parkingSpace) {
```

Comment:
are you using this method in any other place? if not, it should be deleted or be private

Reply:
Done!